examples/tutorials/advanced/grid_equalization.py

```diff
@@ -0,0 +1,223 @@
+"""
+Performing grid histogram equalization
+--------------------------------------
+The :meth:`pygmt.grdhisteq.equalize_grid` function creates a grid with
+statistics based on a cumulative distribution function.
+"""
+# sphinx_gallery_thumbnail_number = 3
+
+import pygmt
+
+###############################################################################
+# Load sample data
+# ----------------
+# Load the sample Earth relief data for a region around Yosemite valley
+# and use :meth:`pygmt.grd2xyz` to create a :class:`pandas.Series` with the
+# z values.
+
+grid = pygmt.datasets.load_earth_relief(
+    resolution="03s", region=[-119.825, -119.4, 37.6, 37.825]
+)
+grid_dist = pygmt.grd2xyz(grid=grid, output_type="pandas")["elevation"]
+
+###############################################################################
+# Plot the original digital elevation model and data distribution
+# ---------------------------------------------------------------
+# For comparison, we will create a map of the original digital elevation
+# model and a histogram showing the distribution of data values.
+
+# Create an instance of the Figure class
+fig = pygmt.Figure()
+# Define figure configuration
+pygmt.config(FORMAT_GEO_MAP="ddd.x", MAP_FRAME_TYPE="plain")
+# Define the colormap for the figure
+pygmt.makecpt(series=[500, 3540], cmap="turku")
+# Define the position of the colormap
+cmap_position = "JMR+o1c/0c+w3c/0.3c"
+# Setup subplots with two panels
+with fig.subplot(
+    nrows=1, ncols=2, figsize=("13.5c", "4c"), title="Digital Elevation Model"
+):
+    # Plot the original digital elevation model in the first panel
+    with fig.set_panel(panel=0):
+        fig.grdimage(grid=grid, projection="M?", frame="WSne", cmap=True)
+    # Plot a histogram showing the z-value distribution in the original digital
+    # elevation model
+    with fig.set_panel(panel=1):
+        fig.histogram(
+            data=grid_dist,
+            projection="X?",
+            region=[500, 3600, 0, 20],
+            series=[500, 3600, 100],
+            frame="wnSE",
```

Comment:
Histogram could use some unit labels

```suggestion
            frame=["wnSE", "xaf+lElevation (m)", "yaf+lCounts"],
```